### DISCUSSION TITLE
katex with dom and html work only after page reload

### DISCUSSION THREAD

**Discussion by smola14** [KaTeX/KaTeX#3475] [Q&A]
Hello, im trying to create math app. I have in javascript objects with question with own id, answer,.. and in the question, i have som katex structures, which render only if i reload page. I have functionality render on page, so when you click on the reload button it change to another question. And here the problem come, that katex doesnt render. Can you help me? Thanks! My repositary: https://github.com/smola14/maturita/. (I have katex only in question with id 4 yet).
Thanks for answer.

**Comment by mbourne** [comment]
Hello @smola14 

You need to call the function `renderMathInElement(elem, options)` at the point in your script when you want KaTeX to render the new math which is loaded when your users click "reload". (This is because after the initial page load when KaTeX renders all it can find, it won't do anything more unless you ask it to.)

If your new math appears in a div with id="newQuestion" (for example), you would call something like `renderMathInElement('newQuestion')`

See some documentation half-way down this page:

https://katex.org/docs/autorender.html

**Reply by edemaine** [reply]
Thanks @mbourne! Small fix: You probably meant `renderMathInElement(document.querySelector('#newQuestion'))` (or `renderMathInElement(document.getElementById('newQuestion'))`).

**Comment by mbourne** [comment]
Oops, indeed. I almost always declare a variable for things like `document.querySelector('#newQuestion')`.

Thanks for the correction.
